sbpy/spectroscopy/sources.py

```diff
@@ -0,0 +1,485 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""sbpy Spectroscopy Sources Module
+
+Spectrophotometric classes that encasulate synphot.SpectralSource and
+synphot.Observation in order to generate sbpy spectra and photometry.
+
+Requires synphot.
+
+"""
+
+__all__ = [
+    'BlackbodySource'
+]
+
+from abc import ABC
+import numpy as np
+import astropy.units as u
+from ..exceptions import SinglePointSpectrumError
+from ..bib import register
+
+
+class SpectralSource(ABC):
+    """Abstract base class for SBPy spectral sources.
+
+    Requires `~synphot`.
+
+    Parameters
+    ----------
+    source : `~synphot.SourceSpectrum`
+        The source spectrum.
+
+    description : string, optional
+        A brief description of the source spectrum.
+
+
+    Attributes
+    ----------
+    wave        - Wavelengths of the source spectrum.
+    fluxd       - Source spectrum.
+    description - Brief description of the source spectrum.
+    meta        - Meta data from ``source``, if any.
+
+    """
+
+    def __init__(self, source, description=None):
+        try:
+            import synphot
+        except ImportError:
+            raise ImportError(
+                'synphot required for {}.'.format(self.__class__.__name__))
```

Comment:
The `__init__` method does not need to import synphot since it will be imported later in the methods that require the source spectrum class.

Reply:
I think the class is useless without synphot, so I wanted to prevent any instantiation at __init__.  Is that acceptable?  I can add a comment to this effect.

Reply:
makes sense!